src/System.Management.Automation/engine/MshCommandRuntime.cs

```diff
@@ -3740,6 +3740,13 @@ internal void SetVariableListsInPipe()
 
```

Comment:
@vexx32, your last commit had 1 failures in `PowerShell-CI-macos`
Parameter Binding Tests.PipelineVariable shouldn't cause a NullRef exception when 'DynamicParam' block is present
```powershell
Cannot retrieve the dynamic parameters for the cmdlet. Object reference not set to an instance of an object.
at <ScriptBlock>, /Users/runner/runners/2.168.2/work/1/s/test/powershell/engine/ParameterBinding/ParameterBinding.Tests.ps1: line 256
256:         DynamicParamTest -PipelineVariable bar | ForEach-Object { $bar } | Should -Be "hi"

```

Reply:
That explains where this bug came from. Someone, at some point, tried to fix that issue and ended up introducing the current bug. This will probably get interesting... I'll do some debugging tomorrow.